packages/databases-collections/src/components/collection-fields/collection-fields.jsx

```diff
@@ -0,0 +1,312 @@
+/* eslint-disable react/no-multi-comp */
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import Checkbox from '@leafygreen-ui/checkbox';
+import TextInput from '@leafygreen-ui/text-input';
+import { Select, Option, Size } from '@leafygreen-ui/select';
+import IconButton from '@leafygreen-ui/icon-button';
+import Icon from '@leafygreen-ui/icon';
+
+import COLLATION_OPTIONS from '../../constants/collation';
+import hasTimeSeriesSupport from './has-time-series-support';
+import styles from './collection-fields.less';
+
+const HELP_URL_CAPPED = 'https://docs.mongodb.com/manual/core/capped-collections/';
+const HELP_URL_COLLATION = 'https://docs.mongodb.com/master/reference/collation/';
+
+const TIME_FIELD_INPUT_DESCRIPTION = 'Specify which field should be used ' +
+  'as timeField for the time-series collection.';
+
+const META_FIELD_INPUT_DESCRIPTION = 'The designated field ' +
+  'for metadata.';
+
+const EXPIRE_AFTER_SECONDS_DESCRIPTION = 'Enables ' +
+  'automatic deletion of documents older than the specified number of seconds.';
+
+class FieldSet extends PureComponent {
+  static propTypes = {
+    children: PropTypes.node
+  }
+
+  render() {
+    return (
+      <fieldset className={styles['form-group']}>
+        {this.props.children}
+      </fieldset>
+    );
+  }
+}
+
+class CollapsibleFieldset extends PureComponent {
+  static propTypes = {
+    children: PropTypes.node,
+    label: PropTypes.string.isRequired,
+    description: PropTypes.oneOfType([
+      PropTypes.element,
+      PropTypes.string
+    ]),
+    helpUrl: PropTypes.string,
+    onToggle: PropTypes.func.isRequired,
+    toggled: PropTypes.bool,
+    openLink: PropTypes.func
+  }
+
+  render() {
+    return (
+      <FieldSet>
+        <Checkbox
+          onChange={event => {
+            this.props.onToggle(event.target.checked);
+          }}
+          label={this.props.label}
+          checked={this.props.toggled}
+          bold={false}
+        />
+        {!this.props.description ? '' : this.props.description}
+        {!!this.props.helpUrl && (
+          <IconButton
+            className={styles['info-btn']}
+            aria-label="Time-series collections documentation"
+            onClick={() => this.props.openLink(this.props.helpUrl)}
+          >
+            <Icon
+              glyph="InfoWithCircle"
+              size="small"
+            />
+          </IconButton>
+        )}
+        {!this.props.toggled ? '' : this.props.children}
+      </FieldSet>
+    );
+  }
+}
+
+export default class CollectionFields extends PureComponent {
```

Comment:
Here's the `CollectionFields` component where we render the options for creating collections (shared w/ create database). 
The more general components above are things we probably want to have in a generalized place in Compass where we can also pull the theme.